pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -1384,6 +1384,9 @@ type WorkflowStep struct {
 	// Hooks holds the lifecycle hook which is invoked at lifecycle of
 	// step, irrespective of the success, failure, or error status of the primary step
 	Hooks LifecycleHooks `json:"hooks,omitempty" protobuf:"bytes,12,opt,name=hooks"`
+
+	// Priority is used if controller is configured to process limited number of workflows in parallel. Steps with higher priority are processed first.
```

Comment:
I think the description here is not quite right,
the priority tag at `step/task` is different to `priority` at the `workflowspec`

Reply:
@tczhao is correct. This should be consistent. Conventionally, N is higher priority than N+1 (i.e. 1 is higher than 2). Which is different to precedence. "Steps with higher priority are processed first." confused me, and made me wonder if it was the reverse. If you follow the convention, delete "Steps with higher priority are processed first.".

Reply:
I think this field should be `int32` not `*int32` as it looks like 0 should be the default.

Reply:
I think the higher priority part is correct. It aligns with how kubernetes prioirtyClass is defined. (The pods with higher priority get scheduled first. e.g. system-cluster-critical class has 2000000000 priority value).

Reply:
To keep in line with k8s priorityClass, I will keep the convention that higher priority = higher value. I will update the comments to accurately reflect that.